cmd/package-image/steps.go

```diff
@@ -0,0 +1,199 @@
+		if err != nil {
+			return p, fmt.Errorf("buildah bud: %w", err)
+		}
+		fmt.Printf("Creating local tar folder for image %s ...\n", p.image.ImageName())
+		err = p.buildahPushTar(os.Stdout, os.Stderr)
+		if err != nil {
+			return p, fmt.Errorf("buildah push tar: %w", err)
+		}
+		d, err := getImageDigestFromFile(p.opts.checkoutDir)
+		if err != nil {
+			return p, err
+		}
+		p.image.Digest = d
+		return p, nil
+	}
+}
+
+func generateSBOM() PackageStep {
+	return func(p *packageImage) (*packageImage, error) {
+		fmt.Println("Generating image SBOM with trivy scanner ...")
+		err := p.generateImageSBOM()
+		if err != nil {
+			return p, fmt.Errorf("generate SBOM: %w", err)
+		}
+		return p, nil
+	}
+}
+
+func scanImageWithAqua() PackageStep {
+	return func(p *packageImage) (*packageImage, error) {
+		if aquasecInstalled() {
+			fmt.Println("Scanning image with Aqua scanner ...")
+			aquaImage := p.image.ImageName()
+			htmlReportFile := filepath.Join(p.opts.checkoutDir, "report.html")
+			jsonReportFile := filepath.Join(p.opts.checkoutDir, "report.json")
+			scanArgs := aquaAssembleScanArgs(p.opts, aquaImage, htmlReportFile, jsonReportFile)
+			scanSuccessful, err := aquaScan(aquasecBin, scanArgs, os.Stdout, os.Stderr)
+			if err != nil {
+				return p, fmt.Errorf("aqua scan: %w", err)
+			}
+
+			if !scanSuccessful && p.opts.aquasecGate {
+				return p, errors.New("stopping build as successful Aqua scan is required")
+			}
+
+			asu, err := aquaScanURL(p.opts, aquaImage)
+			if err != nil {
+				return p, fmt.Errorf("aqua scan URL: %w", err)
+			}
+			fmt.Printf("Aqua vulnerability report is at %s ...\n", asu)
+
+			err = copyAquaReportsToArtifacts(htmlReportFile, jsonReportFile)
+			if err != nil {
+				return p, err
+			}
+
+			fmt.Println("Creating Bitbucket code insight report ...")
+			err = createBitbucketInsightReport(p.opts, asu, scanSuccessful, p.ctxt)
+			if err != nil {
+				return p, err
+			}
+		} else {
+			fmt.Println("Aqua is not configured, image will not be scanned for vulnerabilities.")
+		}
+		return p, nil
+	}
+}
+
+func pushImage() PackageStep {
+	return func(p *packageImage) (*packageImage, error) {
+		fmt.Printf("Pushing image %s ...\n", p.image.ImageName())
+		err := p.buildahPush(os.Stdout, os.Stderr)
+		if err != nil {
+			return p, fmt.Errorf("buildah push: %w", err)
+		}
+		return p, nil
+	}
+}
+
+func storeArtifact() PackageStep {
+	return func(p *packageImage) (*packageImage, error) {
+		err := writeImageDigestToResults(p.image.Digest)
+		if err != nil {
+			return p, err
+		}
+
+		fmt.Println("Writing image artifact ...")
+		imageArtifactFilename := fmt.Sprintf("%s.json", p.image.Name)
+		err = pipelinectxt.WriteJsonArtifact(p.image, pipelinectxt.ImageDigestsPath, imageArtifactFilename)
+		if err != nil {
+			return p, err
+		}
+
+		fmt.Println("Writing SBOM artifact ...")
+		sbomFile := filepath.Join(p.opts.checkoutDir, pipelinectxt.SbomsFilename)
+		if _, err := os.Stat(sbomFile); err == nil {
```

Comment:
I'm confused by this line. Isn't this checking if the SBOM file exists?

I'd think the logic should be the same as for the image artefact: we should always attempt to write here.

Reply:
Also, the name needs to depend on the image name as with the image digest above, also see my comment at `pipelinectxt.SbomsFilename`.